engine/src/main/java/io/zeebe/engine/state/appliers/ProcessInstanceElementActivatingApplier.java

```diff
@@ -53,6 +53,21 @@ public void applyState(final long elementInstanceKey, final ProcessInstanceRecor
 
     if (flowScopeInstance == null) {
       // process instance level
+      final var parentElementInstance =
+          elementInstanceState.getInstance(value.getParentElementInstanceKey());
+      if (parentElementInstance == null) {
+        // root process (not a child process)
+        return;
+      }
+
+      // this check is not really necessary: if parentElementInstance exists,
+      // it should always be a call-activity, but let's try to be safe
```

Comment:
I don't understand why - don't all element instances inside a process have a parent instance which is the process?

Reply:
All element instances inside a process have **process** instance key to refer to the process they belong to. This comment relates to a Process element (not a child element of a process). If it is a child process it should have the call activity as it's **parent** element instance.

Reply:
Ah, true, it's within the `flowScopeInstance == null` check :+1: No all good, my bad